compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -1433,8 +1451,6 @@ class JSCodeGen()(using genCtx: Context) {
 
       case Assign(lhs0, rhs) =>
         val sym = lhs0.symbol
-        if (sym.is(JavaStaticTerm))
-          throw new FatalError(s"Assignment to static member ${sym.fullName} not supported")
```

Comment:
Could we still throw an error if `sym` is not defined in the same compilation unit? Because there is still no ABI for setting static fields. This is only valid within a compilation unit, where we are allowed to use whatever implementation details we want.

Reply:
added a check here